examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py

```diff
@@ -81,28 +57,42 @@ def main():
         MODEL_ID, export=True, load_in_8bit=False, compile=False, stateful=False
     )
 
-    original_ov_model = hf_model.model.clone()
-    evaluator = Evaluator(hf_model, tokenizer=tokenizer, metrics=("similarity",))
-
-    # Wikitext-based compression
-    wikitext_dataset = datasets.load_dataset("wikitext", "wikitext-2-raw-v1", split="test")
-    wikitext_dataset = [d["text"] for d in wikitext_dataset]
-    wikitext_optimized_model = compress_model(hf_model, tokenizer, wikitext_dataset)
+    dataset_size = 100
 
     # Synthetic-based compression
     saved_seed = torch.seed()
     torch.manual_seed(SEED)
-    synthetic_dataset = nncf.data.generate_text_data(hf_model, tokenizer)
+    synthetic_dataset = nncf.data.generate_text_data(hf_model, tokenizer, dataset_size=dataset_size)
+    quantization_dataset = nncf.Dataset(
+        synthetic_dataset, partial(transform_func, tokenizer=tokenizer, ov_model=hf_model.model)
+    )
+    hf_model.request = None
     torch.manual_seed(saved_seed)
-    synthetic_optimized_model = compress_model(hf_model, tokenizer, synthetic_dataset)
 
-    # Similarity comparison between Wikitext-based & Synthetic-based compressed models
-    wikitext_based_similarity = validate_model(evaluator, hf_model, wikitext_optimized_model, original_ov_model)
-    print(f"Wikitext-quantized model similarity: {wikitext_based_similarity}")
+    optimized_model = nncf.compress_weights(
+        hf_model.model.clone(),
+        dataset=quantization_dataset,
+        mode=nncf.CompressWeightsMode.INT4_SYM,
+        ratio=1.0,
+        scale_estimation=True,
+    )
+
+    # Verify the model output in comparison to floating-point one
+    input_ids = tokenizer("What is Python? ", return_tensors="pt").to(device=hf_model.device)
+    max_new_tokens = 100
+
+    fp_output = hf_model.generate(**input_ids, max_new_tokens=max_new_tokens)
+    fp_output_text = tokenizer.decode(fp_output[0])
+
+    print(f"FP model output: {fp_output_text} \n")
+
+    hf_model.model = optimized_model
+    hf_model.request = None
+    opt_output = hf_model.generate(**input_ids, max_new_tokens=max_new_tokens)
+    opt_output_text = tokenizer.decode(opt_output[0])
 
-    synthetic_based_similarity = validate_model(evaluator, hf_model, synthetic_optimized_model, original_ov_model)
-    print(f"Synthetic-quantized model similarity: {synthetic_based_similarity}")
-    return wikitext_based_similarity, synthetic_based_similarity
+    print(f"Optimized model output: {opt_output_text}\n")
+    return opt_output_text
```

Comment:
Could you provide `fp_output_text` and `opt_output_text`?

Reply:
FP model output: What is Python? 

`Python is a high-level, interpreted, general-purpose programming language. It is designed to be easy to learn and use, and it has a large community of developers who contribute to its development. Python is widely used in a variety of fields, including data science, machine learning, web development, and more.`
`Python is a dynamic language, meaning that it dynamically types its code based on the data it receives. This means that Python can automatically detect the type of data being`

Compiling the model to CPU ...
Optimized model output: What is Python? 

`Python is a high-level, interpreted, interactive, and object-oriented programming language. It is widely used in various fields such as data science, artificial intelligence, web development, and machine learning. Python is a popular programming language because it is easy to learn, has a large community, and is widely used in various fields. Python is a dynamic language, which means that it dynamically types the data it receives. This means that Python can dynamically determine the data type of the data it`

Reply:
@MaximProshin, @KodiaqQ I would suggest not to output `fp_output_text`. It only interferes with assessing the correctness of the answer, and leads to an attempt to compare answers word by word.

Reply:
In this case, the example would be useless, for me.
But as you wish.
Done, removed.